pkg/cmd/cli/cmd/rollback.go

```diff
@@ -38,151 +43,267 @@ a human-readable representation of the updated deployment configuration instead
+	cmd.Flags().BoolVar(&opts.IncludeTriggers, "change-triggers", false, "Include the previous deployment's triggers in the rollback")
+	cmd.Flags().BoolVar(&opts.IncludeStrategy, "change-strategy", false, "Include the previous deployment's strategy in the rollback")
+	cmd.Flags().BoolVar(&opts.IncludeScalingSettings, "change-scaling-settings", false, "Include the previous deployment's replicationController replica count and selector in the rollback")
+	cmd.Flags().BoolVarP(&opts.DryRun, "dry-run", "d", false, "Instead of performing the rollback, describe what the rollback will look like in human-readable form")
+	cmd.Flags().StringVarP(&opts.Format, "output", "o", "", "Instead of performing the rollback, print the updated deployment configuration in the specified format (json|yaml|template|templatefile)")
+	cmd.Flags().StringVarP(&opts.Template, "template", "t", "", "Template string or path to template file to use when -o=template or -o=templatefile.")
+	cmd.Flags().IntVar(&opts.DesiredVersion, "to-version", 0, "A config version to rollback to. Specifying version 0 is the same as omitting a version (the version will be auto-detected.)")
 
 	return cmd
 }
 
-// newHelper makes a hew helper using real clients.
-func newHelper(oClient client.Interface, kClient kclient.Interface) *helper {
-	return &helper{
-		generateRollback: func(namespace string, config *deployapi.DeploymentConfigRollback) (*deployapi.DeploymentConfig, error) {
-			return oClient.DeploymentConfigs(namespace).Rollback(config)
-		},
-		describe: func(config *deployapi.DeploymentConfig) (string, error) {
-			describer := describe.NewDeploymentConfigDescriberForConfig(oClient, kClient, config)
-			return describer.Describe(config.Namespace, config.Name)
-		},
-		updateConfig: func(namespace string, config *deployapi.DeploymentConfig) (*deployapi.DeploymentConfig, error) {
-			return oClient.DeploymentConfigs(namespace).Update(config)
-		},
-	}
-}
+type RollbackOptions struct {
+	Namespace              string
+	TargetName             string
+	DesiredVersion         int
+	Format                 string
+	Template               string
+	DryRun                 bool
+	IncludeTriggers        bool
+	IncludeStrategy        bool
+	IncludeScalingSettings bool
 
-// helper knows how to perform various rollback related tasks.
-type helper struct {
-	// generateRollback generates a rolled back config from the input config
-	generateRollback func(namespace string, config *deployapi.DeploymentConfigRollback) (*deployapi.DeploymentConfig, error)
-	// describe returns the describer output for config
-	describe func(config *deployapi.DeploymentConfig) (string, error)
-	// updateConfig persists config
-	updateConfig func(namespace string, config *deployapi.DeploymentConfig) (*deployapi.DeploymentConfig, error)
+	out        io.Writer
+	oc         client.Interface
+	kc         kclient.Interface
+	getBuilder func() *resource.Builder
```

Comment:
This can be turned into a `*resource.Builder`, right?  I don't think the no-arg function is providing anything extra for you.